### PR TITLE
Add count method

### PR DESCRIPTION
**Pull request by sea1jxr**
I was really wishing I had count today, while using genSequence so I added it.

### CODE REVIEW

**Comment by Jason3S** [comment]
Sorry about not getting back to you sooner on this one.

Nicely done with the code.

I have a concern that has nothing to do with your code. The `.count()` will consume the iterator. If the iterator is an array, you can re-use it. But it is not re-usable if the iterator came from a generator.

File: src/GenSequence.test.ts [hunk 1]
```diff
@@ -496,4 +496,19 @@ describe('GenSequence Tests', function() {
...
+
+    it('test count with 1 element', () => {
+        const values: number[] = [1];
+        expect(genSequence(values).count()).to.equal(1);
+    });
+
+    it('test count with 2 elements', () => {
+        const values: number[] = [18, 7];
+        expect(genSequence(values).count()).to.equal(2);
+    });
```
Comment (Jason3S) [inline src/GenSequence.test.ts]: Nicely done. There is two more tests necessary.
Need to test the behavior of what happens if the sequence is used again after `.count()`.

Example:
```typescript
const values = [1, 2, 3, 4, 5, 6]
const seq = genSequence(values);
const cnt = seq.count();
const cntAgain = seq.count();
```
and
```typescript
const values = [1, 2, 3, 4, 5, 6]
const seq = genSequence(values).filter(a => !!(a % 2));
const cnt = seq.count();
const cntAgain = seq.count();
```

Comment (sea1jxr) [inline src/GenSequence.test.ts]: .NET has the same issue with IEnumerables and LINQ, but they allow it, and many times the sequence can handle a reuse.

**Comment by sea1jxr** [comment]
I made sequences reusable.  I ended up removing the next() method because the iterator needed to be reset after it saw done.  I thought maybe that was a bad practice, but the spec doesn't seem to prohibit it, so if there are good scenarios for next() we could bring it back for reusable sequences.

This was kind of an experiment to see if I could make them reusable.  It turned out well, so I thought I would run it by you.

Let me know what you think.

**Comment by Jason3S** [comment]
Wow. Very cool.
You have made it late binding. Similar to RxJs.
I need a bit more time to think about the implications.

I think your suggestion is worth exploring. I will have to bump the major version number, since it is a change to the underlying behavior.

Typing and thinking at the same time (not a good combo for me):
* I think `.next()` is necessary to meet the iterator spec and removing it might break some stuff (not sure).

**Comment by sea1jxr** [comment]
I added back ```.next()```.  Take you time and think through the scenarios.

**Pull request merged**